### ISSUE TITLE
Batch mode tries to query "where in (Null)"

### ISSUE THREAD

**Issue by Venommmm** [src-d/go-kallax#195]
models :

```go
type User struct {
	kallax.Model `table:"users"`
	ID           int64 `pk:"autoincr"`
	Username     string
	Email        string
	Password     string
	CreatedAt    time.Time
	Cards        []*Card       `fk:"user_id"`
	CreditCards  []*CreditCard `fk:"user_id"`
}

type Card struct {
	kallax.Model `table:"cards"`
	ID           int64 `pk:"autoincr"`
	User         *User `fk:"user_id,inverse"`
	Number       string
}

type CreditCard struct {
	kallax.Model `table:"credit_cards"`
	ID           int64 `pk:"autoincr"`
	User         *User `fk:"user_id,inverse"`
	Number       string
}
```

 Query :

```go
q := models.NewUserQuery().WithCards(nil).WithCreditCards(nil)
users, err :=  store.Debug().FindAll(q)
```

Debug output:

```
2017/07/09 14:39:52 kallax: Query: SELECT __user.id, __user.username, __user.email, __user.password, __user.created_at FROM users __user LIMIT 50 OFFSET 0, args: []
2017/07/09 14:39:52 kallax: Query: SELECT __card_Cards.id, __card_Cards.user_id, __card_Cards.number FROM cards __card_Cards WHERE __card_Cards.user_id IN ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15), args: [3 40 35 41 36 42 37 43 1 44 38 45 2 46 39]
2017/07/09 14:39:52 kallax: Query: SELECT __creditcard_CreditCards.id, __creditcard_CreditCards.user_id, __creditcard_CreditCards.number FROM credit_cards __creditcard_CreditCards WHERE __creditcard_CreditCards.user_id IN ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15), args: [3 40 35 41 36 42 37 43 1 44 38 45 2 46 39]
2017/07/09 14:39:52 kallax: Query: SELECT __user.id, __user.username, __user.email, __user.password, __user.created_at FROM users __user LIMIT 50 OFFSET 15, args: []
2017/07/09 14:39:52 kallax: Query: SELECT __card_Cards.id, __card_Cards.user_id, __card_Cards.number FROM cards __card_Cards WHERE __card_Cards.user_id IN (NULL), args: []
2017/07/09 14:39:52 kallax: Query: SELECT __creditcard_CreditCards.id, __creditcard_CreditCards.user_id, __creditcard_CreditCards.number FROM credit_cards __creditcard_CreditCards WHERE __creditcard_CreditCards.user_id IN (NULL), args: []
```

With first query kallax returns just 15 records from table users (with limit 50), so why kallax queries users table again with 15 offset and then queries creditcards, cards with "WHERE user_id IN (NULL) "?

**Comment by erizocosmico** [comment]
Hi, thanks for the report! We'll try to fix this as soon as possible.

The second query with the offset 15 is expected behaviour, if it encounters 0, then the batching stops as it does not know the total amount of results. The `NULL` thing is definitely a bug, it should not have made such query after receiving 0 rows.

**Comment by Venommmm** [comment]
Does kallax really need to continue quering users table again if amount of rows from last query is less than limit?

**Comment by erizocosmico** [comment]
Actually, it wouldn't need to. Will start a separate issue to track that enhancement

**Comment by erizocosmico** [comment]
@Venommmm Can you confirm that this is fixed in v1.2.14?

**Comment by Venommmm** [comment]
Still same logs
2017/07/13 00:44:05 kallax: Query: SELECT __user.id, __user.username, __user.email, __user.password, __user.created_at FROM users __user LIMIT 50 OFFSET 0, args: []
2017/07/13 00:44:05 kallax: Query: SELECT __card_Cards.id, __card_Cards.user_id, __card_Cards.number FROM cards __card_Cards WHERE __card_Cards.user_id IN ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15), args: [3 40 35 41 36 42 37 43 1 44 38 45 2 46 39]
2017/07/13 00:44:05 kallax: Query: SELECT __creditcard_CreditCards.id, __creditcard_CreditCards.user_id, __creditcard_CreditCards.number FROM credit_cards __creditcard_CreditCards WHERE __creditcard_CreditCards.user_id IN ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15), args: [3 40 35 41 36 42 37 43 1 44 38 45 2 46 39]
2017/07/13 00:44:05 kallax: Query: SELECT __user.id, __user.username, __user.email, __user.password, __user.created_at FROM users __user LIMIT 50 OFFSET 15, args: []
2017/07/13 00:44:05 kallax: Query: SELECT __card_Cards.id, __card_Cards.user_id, __card_Cards.number FROM cards __card_Cards WHERE __card_Cards.user_id IN (NULL), args: []
2017/07/13 00:44:05 kallax: Query: SELECT __creditcard_CreditCards.id, __creditcard_CreditCards.user_id, __creditcard_CreditCards.number FROM credit_cards __creditcard_CreditCards WHERE __creditcard_CreditCards.user_id IN (NULL), args: []

Kallax v 1.2.14
